examples/dataflow-cloud-run/dataflowApp/src/main/java/com/demo/dataflow/eventfn/ParseGoBikeEvents.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.demo.dataflow.eventfn;
+
+import com.demo.dataflow.model.FailedMessage;
+import com.demo.dataflow.model.GoBike;
+import com.demo.dataflow.util.ErrorCodes;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.TupleTagList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.UUID;
+
+
+public class ParseGoBikeEvents extends DoFn<String, GoBike> {
+
+    public static  String headers = "\"duration_sec\",\"start_time\",\"end_time\",\"start_station_id\",\"start_station_name\",\"start_station_latitude\",\"start_station_longitude\",\"end_station_id\",\"end_station_name\",\"end_station_latitude\",\"end_station_longitude\",\"bike_id\",\"user_type\",\"member_birth_year\",\"member_gender\",\"bike_share_for_all_trip\"";
+    // Log and count parse errors.
+    private static final Logger LOG = LoggerFactory.getLogger(ParseGoBikeEvents.class);
+
+    public static TupleTag<GoBike> successTag = new TupleTag<GoBike>() {};
+    public static TupleTag<FailedMessage> deadLetterTag = new TupleTag<FailedMessage>() {};
+
+    public static PCollectionTuple process(PCollection<String> csvLines, Counter successCounter, Counter failureCounter) {
+        return csvLines.apply("Parse PDC Data and Extract Information", ParDo.of(new DoFn<String, GoBike>() {
+            @ProcessElement
+            public void processElement(ProcessContext c) {
+                if (!c.element().toString().equals(headers)) {
+                    try {
+                        String[] components = c.element().toString().split(",");
+                        int durationSec = Integer.parseInt(components[0].replace("\"", ""));
```

Comment:
A lot of code duplication going on here and parsing CSVs by hand is dangerous.
Instead use something standards based like [CSVParser](https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVParser.html) to achieve this in a much more concise manner.

It's hard to trust code like this without tests :)

Reply:
I have incorporated the requested change